packages/x-utils/tsconfig.json

```
@@ -0,0 +1,17 @@
+{
+  "compilerOptions": {
+    "target": "es2019",
+    "module": "commonjs",
+    "outDir": "./dist/cjs",
+    "declaration": true,
```

Add `declarationDir` property to generate `.d.ts` files under `/types` folder

```suggestion
    "declaration": true,
    "declarationDir": "./types"
```